### PR TITLE
feat(framework) Add secure channel support for SuperExec

### PR DESCRIPTION
**Pull request by charlesbvll**
### Checklist

- [ ] Implement proposed change
- [ ] Write tests
- [ ] Update [documentation](https://flower.ai/docs/writing-documentation.html)
- [ ] Make CI checks pass
- [ ] Ping maintainers on [Slack](https://flower.ai/join-slack/) (channel `#contributions`)

### CODE REVIEW

File: src/py/flwr/cli/run/run.py [hunk 1]
```diff
@@ -102,10 +102,17 @@ def on_channel_state_change(channel_connectivity: str) -> None:
         """Log channel connectivity."""
         log(DEBUG, channel_connectivity)
 
+    if cert_path := federation.get("root-certificates"):
+        root_certificates = Path(cert_path).read_bytes()
```
Comment (danieljanes) [inline src/py/flwr/cli/run/run.py:106]: In other parts of the codebase, we use `root_certificates` to refer to the path, and `root_certificates_bytes` to refer to the content

Comment (charlesbvll) [reply src/py/flwr/cli/run/run.py:106]: Sounds good, I'll update it

File: src/py/flwr/cli/run/run.py [hunk 2]
```diff
@@ -102,10 +102,17 @@ def on_channel_state_change(channel_connectivity: str) -> None:
         """Log channel connectivity."""
         log(DEBUG, channel_connectivity)
 
+    if root_certificates := federation.get("root-certificates"):
+        root_certificates_bytes = Path(root_certificates).read_bytes()
+        insecure = False
+    else:
+        root_certificates_bytes = None
+        insecure = True
```
Comment (danieljanes) [inline src/py/flwr/cli/run/run.py:110]: In both cases, we need to read an `insecure` flag from the `federation` config.

If `root_certificates` are provided, it's ok if the user omits it. If the user sets it, it must be set to `false`.

If `root_certificates` are not provided, then the user must provide it and it must be set to true.

Comment (charlesbvll) [reply src/py/flwr/cli/run/run.py:126]: Btw @danieljanes , should those settings be under `[tool.flwr.federations.name]` or under `[tool.flwr.federations.name.options]`?

File: src/py/flwr/cli/run/run.py [hunk 3]
```diff
@@ -120,10 +120,40 @@ def on_channel_state_change(channel_connectivity: str) -> None:
...
+        root_certificates_bytes = Path(root_certificates).read_bytes()
+        if bool(insecure_str):
+            typer.secho(
+                "❌ `root_certificates` were provided but the `insecure` parameter"
+                "is set to `True`.",
+                fg=typer.colors.RED,
+                bold=True,
+            )
+            raise typer.Exit(code=1)
+        insecure = False
```
Comment (danieljanes) [inline src/py/flwr/cli/run/run.py:134]: This ling won't be necessary if we remember the result of `bool(insecure_str)` above

File: src/py/flwr/cli/run/run.py [hunk 4]
```diff
@@ -120,10 +120,40 @@ def on_channel_state_change(channel_connectivity: str) -> None:
...
+    else:
+        root_certificates_bytes = None
+        if insecure_str is None:
+            typer.secho(
+                "❌ The insecure parameter must be set in the `pyproject.toml`.",
```
Comment (danieljanes) [inline src/py/flwr/cli/run/run.py:139]:
```suggestion
                "❌ To disable TLS, set `insecure = true` in `pyproject.toml`.",
```

File: src/py/flwr/cli/run/run.py [hunk 5]
```diff
@@ -120,10 +120,40 @@ def on_channel_state_change(channel_connectivity: str) -> None:
...
+            )
+            raise typer.Exit(code=1)
+        if not bool(insecure_str):
+            typer.secho(
+                "❌ No certificate were given yet `insecure` is set to `False`.",
+                fg=typer.colors.RED,
+                bold=True,
+            )
+            raise typer.Exit(code=1)
+        insecure = True
```
Comment (danieljanes) [inline src/py/flwr/cli/run/run.py:151]: Instead of setting this manually, can we remember the result of `bool(insecure_str)`?

**Review by danieljanes — APPROVED** [review]

**Pull request merged**